### ISSUE TITLE
5.x kernel headers now required due to `seccomp_notif_sizes`

### ISSUE THREAD

**Issue by loganwendholt** [rr-debugger/rr#2907]
21cad8366c7470ac62e5bbb89a7ec03da0ae3302 performs a type check on `seccomp_notif_sizes` from `<linux/seccomp.h>`. This type was added in kernel version 5.0, and does not exist in 4.x or earlier.

Current requirements for `rr` are advertised as kernel ≥ 3.11.

Hopefully this seccomp stuff can be handled in a backwards-compatible way rather than keeping this new implicit kernel ≥ 5.0 requirement.

**Comment by Keno** [comment]
It's using a private definition of that struct: https://github.com/rr-debugger/rr/commit/21cad8366c7470ac62e5bbb89a7ec03da0ae3302#diff-86b09f63e018900202683e278750540fe73a4980e6e0b819df6e4d93bd668a26R626. There should be no problem continuing to run rr on older kernels. If you run into any, please do file an issue about that.

**Comment by loganwendholt** [comment]
Sorry, should have added this to my initial comment:

With 4.x kernel headers, my build fails. Seems to be due to the type check that tries to compare the private definition against the one defined in the actual kernel headers:
```
[ 97%] Building CXX object CMakeFiles/rr.dir/src/ProcMemMonitor.cc.o
/home/lwendholt/rr/src/kernel_abi.h:631:18: error: '::seccomp_notif_sizes' has not been declared
  631 |   RR_VERIFY_TYPE(seccomp_notif_sizes);
      |                  ^~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:66:33: note: in definition of macro 'RR_VERIFY_TYPE_ARCH'
   66 |   static_assert(Verifier<arch_, system_type_, rr_type_>::same_size,            \
      |                                 ^~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:74:31: note: in expansion of macro 'RR_VERIFY_TYPE_EXPLICIT'
   74 | #define RR_VERIFY_TYPE(type_) RR_VERIFY_TYPE_EXPLICIT(::type_, type_)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.h:631:3: note: in expansion of macro 'RR_VERIFY_TYPE'
  631 |   RR_VERIFY_TYPE(seccomp_notif_sizes);
      |   ^~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.h:631:18: error: '::seccomp_notif_sizes' has not been declared
  631 |   RR_VERIFY_TYPE(seccomp_notif_sizes);
      |                  ^~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:66:33: note: in definition of macro 'RR_VERIFY_TYPE_ARCH'
   66 |   static_assert(Verifier<arch_, system_type_, rr_type_>::same_size,            \
      |                                 ^~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:74:31: note: in expansion of macro 'RR_VERIFY_TYPE_EXPLICIT'
   74 | #define RR_VERIFY_TYPE(type_) RR_VERIFY_TYPE_EXPLICIT(::type_, type_)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.h:631:3: note: in expansion of macro 'RR_VERIFY_TYPE'
  631 |   RR_VERIFY_TYPE(seccomp_notif_sizes);
      |   ^~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:66:55: error: template argument 2 is invalid
   66 |   static_assert(Verifier<arch_, system_type_, rr_type_>::same_size,            \
      |                                                       ^
/home/lwendholt/rr/src/kernel_abi.cc:71:3: note: in expansion of macro 'RR_VERIFY_TYPE_ARCH'
   71 |   RR_VERIFY_TYPE_ARCH(arch_, system_type_, rr_type_)
      |   ^~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.cc:74:31: note: in expansion of macro 'RR_VERIFY_TYPE_EXPLICIT'
   74 | #define RR_VERIFY_TYPE(type_) RR_VERIFY_TYPE_EXPLICIT(::type_, type_)
      |                               ^~~~~~~~~~~~~~~~~~~~~~~
/home/lwendholt/rr/src/kernel_abi.h:631:3: note: in expansion of macro 'RR_VERIFY_TYPE'
  631 |   RR_VERIFY_TYPE(seccomp_notif_sizes);
      |   ^~~~~~~~~~~~~~
```

**Comment by Keno** [comment]
Ah, yes that does want newer kernel headers, though it should still run on old kernels if built that way. That said, we could probably either delete that line or use some C++ alchemy to detect whether or not the struct is defined in the global namespace and only check the size conditionally.

**Comment by rocallahan** [comment]
Fixed by 782537e770f31b32272b4d1d1fa4c2ca790cf220. I just removed the check; `seccomp_notif_sizes` is not a very interesting struct to check the size of anyway since it's the same on all platforms.
